geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/command/ExecuteRegionFunction66.java

```diff
@@ -180,61 +153,37 @@ public void cmdExecute(final Message clientMessage, final ServerConnection serve
     try {
       if (function instanceof String) {
         functionObject = FunctionService.getFunction((String) function);
-        if (functionObject == null) {
-          String message =
-              String.format("The function, %s, has not been registered",
-                  function);
-          logger.warn("{}: {}", serverConnection.getName(), message);
-          sendError(hasResult, clientMessage, message, serverConnection);
+        if (!validateFunctionObject(clientMessage, serverConnection, function, hasResult,
+            functionState,
+            functionObject)) {
           return;
```

Comment:
Earlier if we did not satisfy the condition "functionStateOnServerSide != functionState" we did not return from the validation part. Now it seems we return to the caller.